go.mod

```diff
@@ -1,13 +1,14 @@
 module github.com/hewlettpackard/hpegl-metal-terraform-resources
 
-go 1.21.0
+go 1.22.0
```

Comment:
@neilgierman, please check if this is Ok? It is coming after "go mod tidy".

Reply:
I have another PR at https://github.com/HewlettPackard/hpegl-metal-terraform-resources/pull/231 to be a more complete update to gp 1.22. Once that is approved, then the new master can be merged to this PR.